scalalib/api/src/mill/scalalib/api/ZincWorkerApi.scala

```diff
@@ -49,7 +49,25 @@ trait ZincWorkerApi {
       incrementalCompilation = true
     )
 
+  def compileMixed(
+      upstreamCompileOutput: Seq[CompilationResult],
+      sources: Agg[os.Path],
+      compileClasspath: Agg[os.Path],
+      javacOptions: Seq[String],
+      scalaVersion: String,
+      scalaOrganization: String,
+      scalacOptions: Seq[String],
+      compilerClasspath: Agg[PathRef],
+      scalacPluginClasspath: Agg[PathRef],
+      reporter: Option[CompileProblemReporter],
+      reportCachedProblems: Boolean,
+      incrementalCompilation: Boolean,
+      auxiliaryClassFileExtensions: Seq[String],
+      usePipeling: Boolean
+  )(implicit ctx: ZincWorkerApi.Ctx): mill.api.Result[CompilationResult] = ???
```

Comment:
It's probably time to use a parameter class instead of adding new parameters and deprecated `def`s. That way, we only need to extends the API in one place. We could also move validation into that class, if needed.